From f1488bbb0991f99d823d384b00f6fb1de385baa3 Mon Sep 17 00:00:00 2001
From: Aaron Conole <aconole@redhat.com>
Date: Wed, 10 Nov 2021 16:40:20 -0500
Subject: [PATCH 5/8] basman: use return address when pulling address

The managed address pulling routine will fail to reset the return
value from a previous attempt if no IPv4 and IPv6 addresses are
available.  Use the return address of the hwaddr fetch.

Resolves: https://github.com/intel/openlldp/issues/82
Signed-off-by: Aaron Conole <aconole@redhat.com>
---
 lldp_basman.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/lldp_basman.c b/lldp_basman.c
index 25e7d9e..cb0c50c 100644
--- a/lldp_basman.c
+++ b/lldp_basman.c
@@ -515,7 +515,7 @@ static int basman_bld_manaddr_tlv(struct basman_data *bd,
 	if (rc) {
 		rc = basman_get_manaddr_sub(bd, agent, MANADDR_IPV6);
 		if (rc)
-			basman_get_manaddr_sub(bd, agent, MANADDR_ALL802);
+			rc = basman_get_manaddr_sub(bd, agent, MANADDR_ALL802);
 	}
 out_err:
 	return rc;
-- 
2.31.1

